lib/Imgur/Middleware/ErrorMiddleware.php

```diff
@@ -71,9 +71,15 @@ public function checkError(ResponseInterface $response)
         }
 
         if (\is_array($responseData) && isset($responseData['data']) && isset($responseData['data']['error'])) {
-            $message = 'Request failed with: "' . $responseData['data']['error'] . '"';
+            $errorMessage = $responseData['data']['error'];
+            // when uploading too fast, error is an array
+            if (\is_array($responseData['data']['error']) && isset($responseData['data']['error']['message'])) {
```

Comment:
I'm curious about this condition.

Is it possible that the `$responseData['data']['error']['message']` is not set?

If it's possible, the `$responseData['data']['error']` will be the array. And it will be the `Notice message about Array to string conversion`.

Consider following approach:

```php
......
 if (\is_array($responseData['data']['error']) {
    ......
    $errorMessage = $responseData['data']['error']['message'] ?? '';
    ......
}
......
```

It will avoid the Notice message correctly :).

Reply:
I don't know because I do not have other cases where the `error` key is an array. Maybe it can be an array without the `message` key. We don't know.

Also, using `??` is PHP 7.? but the project still needs to work on PHP 5.6 :)

Reply:
@j0k3r, thank you for your reply.

To be compatible with `php-5.6`, perhaps we can consider following code snippet :) :

```php
$errorMessage = isset($responseData['data']['error']['message']) ? $responseData['data']['error']['message'] : '';
```